Bluejay.asm

```diff
@@ -2912,17 +3033,30 @@ dshot_tlm_create_packet:
 	addc	A, Temp2
 	mov	Tlm_Data_H, A
 
+dshot_tlm_ready:
 	Early_Return_Packet_Stage 1
-	mov	A, Tlm_Data_H
+
+	; If extended telemetry ready jump to extended telemetry coded
+	mov A, Ext_Telemetry_H
+	jnz dshot_tlm_ext_coded
 
 	; 12-bit encode telemetry data
+	mov	A, Tlm_Data_H
 	jnz	dshot_12bit_encode
 	mov	A, Tlm_Data_L				; Already 12-bit
 	jnz	dshot_tlm_12bit_encoded
 
 	; If period is zero then reset to FFFFh (FFFh for 12-bit)
 	mov	Tlm_Data_H, #0Fh
 	mov	Tlm_Data_L, #0FFh
+	sjmp dshot_tlm_12bit_encoded
+
+dshot_tlm_ext_coded:
+	; Move extended telemetry data to telemetry data to send
+	mov Tlm_Data_L, Ext_Telemetry_L
+	mov Tlm_Data_H, Ext_Telemetry_H
+	; Clear extended telemetry data
+	mov Ext_Telemetry_H, #0
 
```

Comment:
Only clearing Ext_Telemetry_H and not _L because L will always be written to?

Reply:
Ext_Telemetry_H has double purpose, on one hand it carries the high part (ext. telemetry type) of the extended telemetry value. On the other hand when extended telemetry is checked to be sent this value is compared to 0, and if different, extended telemetry is sent instead of erpm coded value. For that reason setting it to 0 after sending extended telemetry disables sending it again. Ext_Telemetry_L is left as vestigial value, and nothing is done with that variable.